diag_manager/fms_diag_output_buffer.F90

```diff
@@ -295,15 +295,15 @@ subroutine allocate_buffer_0d(this, buff_type, field_name, diurnal_samples)
       allocate(integer(kind=i4_kind) :: this%buffer(1))
       allocate(integer(kind=i4_kind) :: this%counter(1))
       allocate(integer(kind=i4_kind) :: this%count_0d(n_samples))
-      this%counter = 0
-      this%count_0d = 0
+      this%counter = 0_i4_kind
+      this%count_0d = 0_i4_kind
       this%buffer_type = i4
     type is (integer(kind=i8_kind))
       allocate(integer(kind=i8_kind) :: this%buffer(1))
       allocate(integer(kind=i8_kind) :: this%counter(1))
       allocate(integer(kind=i8_kind) :: this%count_0d(n_samples))
-      this%counter = 0
-      this%count_0d = 0
+      this%counter = 0_i8_kind
+      this%count_0d = 0_i8_kind
```

Comment:
Are we specifying the type in the counter integer?  This seems like a waste of space, and could require an implicit cast if it is used or updated by something that is declared as `integer` and the integers are compiled as i4.